src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -88,6 +88,10 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         /* Note: Note the 'try with resource' statement below.
          * More info: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
```

Comment:
- Glad you picked save() to do the "does the file exists" check

- throwing the exception at the end will ensure that the program recovers and warns the user instead of a runtime crash.